### ISSUE TITLE
Provide backup for other WhisperSystems apps

### ISSUE THREAD

**Issue by phipolis** [WhisperSystems/Flock#87]
Creating this as a feature request to correspond with TextSecure issues [TS#1876](https://github.com/WhisperSystems/TextSecure/issues/1876) and [TS#1705](https://github.com/WhisperSystems/TextSecure/issues/1705):

It would be super fantastic if Flock was a backup provider for the settings and encryption keys of TextSecure (et al).

TS' encrypted backup feature has been broken for a long time. As a result, getting a new phone for most users means creating new keys and getting their contacts to accept them. Not only are multiple awkward steps necessary before communication can resume, users get used to being asked to approve random key changes with a "hey i got a new phone" message.

With Flock preserving a user's keys and their contacts' keys, a fresh install of TS could ask if they've used Flock before. If yes, the user could be directed to install Flock, thereby allowing the old keys to be restored and communication to resume seamlessly. Everyone is joyful!

**Comment by manhole11** [comment]
WANT!

**Comment by generalmanager** [comment]
I agree that the Flock technology and codebase could be useful for that, but it would IMHO make a lot more sense to put the relevant parts in a library and use this from within TextSecure, RedPhone and Signal.
There is no point in cluttering a universal CalDAV/CarDAV encryption app to make it more useful for the small minority of users of secure communications apps by OWS.
